### PR TITLE
Remove win32-ia32 target

### PR DESCRIPTION
**Pull request by joaomoreno**
This not only removes Windows 32-bit as a VS Code target, but ia32 support altogether since win32-ia32 was the last ia32 target supported. The last 32-bit target supported is linux-armhf.

To do before merging:
- [x] Test CI & build pipelines
- [x] Merge distro and update distro commit

Would be great to get the following eyes on this:

- @sandy081, for platform specific extension changes
- @connor4312, for CLI changes
- @roblourens, for that small Search change

### CODE REVIEW

**Review by connor4312 — DISMISSED** [review]
Long live 64 bit 🔥 🔥

**Review by roblourens — DISMISSED** [review]
Gonna tell the grandkids that I worked on 32-bit software

**Review by sandy081 — CHANGES_REQUESTED** [review]

**Comment by sandy081** [comment]
@joaomoreno Should we also drop the support for `win32-ia32` target for platform specific extensions also?

**Comment by joaomoreno** [comment]
> @joaomoreno Should we also drop the support for `win32-ia32` target for platform specific extensions also?

Yes, this will follow suit.

File: src/vs/platform/extensionManagement/common/extensionManagement.ts [hunk 1]
```diff
@@ -153,8 +148,8 @@ export function isTargetPlatformCompatible(extensionTargetPlatform: TargetPlatfo
 
 export function getFallbackTargetPlarforms(targetPlatform: TargetPlatform): TargetPlatform[] {
```
[the remark this answers is not in the document]
Comment (joaomoreno) [reply src/vs/platform/extensionManagement/common/extensionManagement.ts:149]: I see usages:

image

[the remark this answers is not in the document]
Comment (sandy081) [reply src/vs/platform/extensionManagement/common/extensionManagement.ts]: Removed it and adopted the usages.

**Review by sandy081 — DISMISSED** [review]

**Review by sandy081 — APPROVED** [review]

**Pull request merged**